src/clean-core/to_string.cc

```diff
@@ -473,265 +473,159 @@ char const* unsigned_int_to_binary(IntType value, char* end)
 template <class IntType>
 void unsigned_to_string_impl(cc::string_stream_ref ss, IntType value, parsed_fmt_args const& args)
 {
+    auto const add_fill = [&](int length) {
+        auto const fill = args.sign_aware_zero_padding ? '0' : args.fill;
+        if (args.width > 0 && length < args.width)
+            for (auto i = 0; i < args.width - length; ++i)
+                ss << fill;
+    };
+
+    auto const to_binary = [&]() {
+        char buffer[sizeof(IntType) * 8 + 1];
+        char* end = buffer + sizeof(buffer);
+        char const* begin = unsigned_int_to_binary(value, end);
+        ss << cc::string_view(begin, end);
+    };
+
     static_assert(std::is_unsigned_v<IntType>);
     switch (args.type)
     {
     case 0:   // fallthrough
     case 'd': // default, decimal
     {
+        auto const buffer_size = [] {
+            switch (sizeof(IntType))
+            {
+            case 1:
+                return 3 + 1; // 255
+            case 2:
+                return 6 + 1; // 65535
+            case 4:
+                return 10 + 1; // 4 294 967 295
+            case 8:
+                return 20 + 1; // 18 446 744 073 709 551 615
+            }
+        }();
+
+        char buffer[buffer_size];
```

Comment:
this is neat but i think a simple 21B buffer in all cases is better here

Reply:
Is there any reason, *why* it's better?

Reply:
I would make buffer_size `constexpr` because I'm not sure if the current `const` is technically ok. Otherwise I think this solution is fine, it's a compile time constant, not a runtime sized buffer

Reply:
Good point, I'll change that just to be sure. However, I believe this is fine. I played around with this for a while, and for some other configurations I got a warning, that variable buffer sizes are not ISO c++.

Reply:
why it would be better: less code, less complexity, less potential for error (ie. those digit counts), fewer potential constexpr-related compiler issues. saving single digit amounts of bytes on the stack isn't worth the trouble is all im saying.

Reply:
I simplified this, as the performance gain is likely non-existent.